package NYT.NObjectClient.NProto;

import "yt/core/misc/guid.proto";
import "yt/core/rpc/rpc.proto";

////////////////////////////////////////////////////////////////////////////////

// Describes a transaction that must be alive in order for the request to be executed.
message TPrerequisiteTransaction
{
    required NYT.NProto.TGuid transaction_id = 1;
}

// Describes node revision that must match in order for the request to be executed.
message TPrerequisiteRevision
{
    required string path = 1;
    required NYT.NProto.TGuid transaction_id = 2;
    required int64 revision = 3;
}

message TPrerequisitesExt
{
    extend NRpc.NProto.TRequestHeader
    {
        optional TPrerequisitesExt prerequisites_ext = 109;
    }

    repeated TPrerequisiteTransaction transactions = 1;
    repeated TPrerequisiteRevision revisions = 2;
}

////////////////////////////////////////////////////////////////////////////////

message TReqGetBasicAttributes
{ }

message TRspGetBasicAttributes
{
    required NYT.NProto.TGuid object_id = 1;
    required int32 cell_tag = 2;
}

////////////////////////////////////////////////////////////////////////////////

message TReqCheckPermission
{
    required string user = 1;
    required int32 permission = 2;
}

message TRspCheckPermission
{
    required int32 action = 1;
    optional NYT.NProto.TGuid object_id = 2;
    optional string object_name = 5;
    optional NYT.NProto.TGuid subject_id = 4;
    optional string subject_name = 3;
}

////////////////////////////////////////////////////////////////////////////////